Update naming convention of galaxy postgres disk [IA-2740]

ehanna4 opened the pull request:
Updated the naming convention for the postgres/metadata disk for Galaxy Apps to use the datadisk name so that the Disk table in the Leo DB will have access to the info needed for resource validator to delete both disks from Google.

I tested with a fiab and Gabriela's UI changes the following successfully:
- Create a Galaxy App with develop branch then switch to my changes and successfully - Delete the App and Disks (checked Google and DB), Delete just the App and keep the Disks and then attach said Disks to a new App
- Create a fully new Galaxy App with the new styled named Disk and successfully deleted this as well.



---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

ehanna4 committed:
Update naming of metadatadisk to use datadisk name

This way the metadatadisk name can be built using the Data table in the DB since the old naming convention pieces were not persisted to the Disk table this wasn't possible previously

ehanna4 committed:
strange name refactoring error fix

ehanna4 committed:
New naming convention works

ehanna4 committed:
Update unittest

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
Need logs for debugging

ehanna4 committed:
Fix log messages (scala is hard)

ehanna4 committed:
Adding backwards compatibility for naming of postgresdisks

ehanna4 committed:
remove debugging logs

Qi77Qi commented on http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/MonitorAtBoot.scala:
```diff
@@ -144,11 +144,13 @@ class MonitorAtBoot[F[_]: Timer](publisherQueue: fs2.concurrent.Queue[F, LeoPubs
             action,
             app.id,
             app.appName,
+            diskNameOpt,
```
why is disk `DiskName` needed to be passed in the message? I think `diskId` is already available from the message, from which you can get the diskName

ehanna4 replied:
I'm not sure i understand how to get `diskName` from `diskId` - wouldn't that require querying the DB?

Qi77Qi replied:
yea..I think in subscriber, we're probably already querying for the disk anyway

ehanna4 replied:
alright, looking into it!

ehanna4 replied:
i think i've made the right change! please let me know what you think!

Qi77Qi commented on http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:
```diff
@@ -1234,7 +1279,7 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
       raw"""galaxy.postgresql.galaxyDatabasePassword=${config.galaxyAppConfig.postgresPassword}""",
       raw"""persistence.postgres.persistentVolume.extraSpec.gcePersistentDisk.pdName=${getGalaxyPostgresDiskName(
-        namespaceName
+        nfsDisk.name
```
we might need to the disk name depending on which style disk actually exists in google if we're worried about users might have an older style disk....but at the same time, Gabriela's UI restore PR is not merged yet, so we probably don't really need to worry about this much

ehanna4 committed:
Address feedback on backwards compat

ehanna4 committed:
Update for correcting postgresname handling

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
Fixing up unit tests p1

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
Some unittest fixings

ehanna4 committed:
Merge branch 'eh-update-postgres-disk-naming' of https://github.com/broadinstitute/leonardo into eh-update-postgres-disk-naming

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
small fixes

ehanna4 committed:
Fixed GKEInterp spec

ehanna4 committed:
trying to fix tests

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
Qi is the best

the subscriber relies on a specific behavior of google disk service, which is mocked by makeDetachingDiskInterp() not MockGoogleDiskService

ehanna4 committed:
Merge branch 'eh-update-postgres-disk-naming' of https://github.com/broadinstitute/leonardo into eh-update-postgres-disk-naming

ehanna4 committed:
Remove disk name from being passed around

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

rtitle commented on http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:
```diff
@@ -896,6 +941,14 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
         s"Installing helm chart ${config.galaxyAppConfig.chart} for app ${appName.value} in cluster ${dbCluster.getGkeClusterId.toString}"
       )
+      postgresDiskNameOpt <- for {
+        disk <- getGalaxyPostgresDisk(nfsDisk.name, namespaceName, nfsDisk.googleProject, nfsDisk.zone)
```
Do we need to call `getGalaxyPostgresDisk` (which makes a Google call) here? Since we just need the name could we do:
```
postgresDiskName = getGalaxyPostgresDiskName(nfsDisk.name)
```

Qi77Qi replied:
this is to support disks that're created in old naming convention...altho I'm not sure how important that is given restore from a disk is not available in UI yet

rtitle replied:
Right, I see… yeah technially disk support is not in the UI yet but probably safe to have this here I suppose.

ehanna4 replied:
yeah I wasn't sure if my change or Gabriela's was going to get out first so figured better safe than sorry lol and will protect against any folks who maybe made a PD and stopped using it awhile will be able to start up again no problem once those UI changes do go out.

rtitle commented on http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriberSpec.scala:
```diff
@@ -1280,7 +1281,7 @@ class LeoPubsubMessageSubscriberSpec
       getCluster.nodepools.size shouldBe 2
       getCluster.nodepools.filter(_.isDefault).head.status shouldBe NodepoolStatus.Running
-      getApp.app.errors.size shouldBe 1
+//      getApp.app.errors.size shouldBe 1 // <- emily error here gets 0
```
Noting TODO, should revert before merge (or understand why it's failing and fix)

rtitle commented on http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:
```diff
@@ -819,19 +823,60 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
+  private[leonardo] def getGalaxyPostgresDiskName(dataDiskName: DiskName): DiskName =
+    DiskName(s"${dataDiskName.value}-${config.galaxyDiskConfig.postgresDiskNameSuffix}")
+
+  private[leonardo] def getOldStyleGalaxyPostgresDiskName(namespaceName: NamespaceName): DiskName =
```
Maybe should add a TODO to delete the `oldStyle` method once there are no legacy Galaxy disks in the system?

Qi77Qi replied:
yea...I think that'd be a good ticket to do

rtitle replied:
Ticket sounds good 👍

ehanna4 replied:
I'll make a ticket👍

rtitle commented on http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala:
```diff
@@ -894,13 +899,9 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
       task = for {
```
Out of scope for this PR, but I recall thinking this method could be cleaned up

ehanna4 replied:
Should I make a note in the code like a TODO or a ticket?

rtitle replied:
Eh, not sure what concretely to suggest so don't worry about it :)

Qi77Qi reviewed (approved):
lgtm

rtitle reviewed (approved):
Few comments above but overall looks good. The testing performed sounds good too. 👍

It would be good to have a follow-up ticket to remove support for legacy disk names, and also cleaning up some of the disk-related code (which existed before this PR).

ehanna4 committed:
uncomment test

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming

ehanna4 committed:
typo in test

ehanna4 committed:
Merge branch 'eh-update-postgres-disk-naming' of https://github.com/DataBiosphere/leonardo into pr/2069

ehanna4 committed:
correct test typo

ehanna4 committed:
Merge branch 'develop' into eh-update-postgres-disk-naming
